Angular/src/app/admin-table/admin-table.component.ts

```diff
@@ -17,109 +17,80 @@ export class AdminTableComponent implements OnInit {
+    })
   }
 
-  async openRoleInfoModal(selectedRoleId) {
-    await this.dataService.getRoleInformation(selectedRoleId).then(response => {
-      const modalRef = this.modalService.open(RoleInformationComponent);
-      modalRef.componentInstance.roleToDisplay = response[0];
-    });
-  }
-
-  async openBandInfoModal(selectedBandId) {
-    await this.dataService.getBandInformation(selectedBandId).then(response => {
-      const modalRef = this.modalService.open(BandInformationComponent);
-      modalRef.componentInstance.bandToDisplay = response[0];
+  openEditModal(roleId, capabilities, bands){
+    this.modalService.openEditRoleModal(roleId, capabilities, bands).then(data => {
+      this.displayAlert(data)
     });
   }
 
   loadRoles() {
-    this.dataService.getCapabilityNamesByDepartment(this.id).then(response => {
+    this.dataService.getCapabilityNamesByDepartment(this.departmentId).then(response => {
       this.capabilities = response;
     });
 
-    this.dataService.getRolesInDepartment(this.id).then(response => {
+    this.dataService.getRolesInDepartment(this.departmentId).then(response => {
       this.jobsInDep = response;
     });
 
     this.dataService.getBands().then(response => {
       this.bands = response;
     });
 
-    this.dataService.getDepartmentDetails(this.id).then(response => {
+    this.dataService.getDepartmentDetails(this.departmentId).then(response => {
       this.departmentName = response;
     });
   }
 
-  openAddRoleModal(roleBandId, roleCapabilityId) {
-    const modalRef = this.modalService.open(AddRoleModalComponent);
-    modalRef.componentInstance.departmentId = this.id;
-    modalRef.componentInstance.bandId = roleBandId;
-    modalRef.componentInstance.capabilityId = roleCapabilityId;
-    modalRef.componentInstance.roleAdded.subscribe(data => {
-    this.showAlert = true;
-      if (data.data.hasOwnProperty('success')) {
-        this.alertMessage = data.data.success;
-        this.alertType = "success";
-      } else {
-        this.alertMessage = data.data.error;
-        this.alertType = "danger";
-      }
-      this.loadRoles();
-    });
-  }
-
-  async openEditRoleModal(roleId) {
-    await this.dataService.getEditRole(roleId).then(response => {
-      const modalRef = this.modalService.open(EditRoleModalComponent);
-      modalRef.componentInstance.roleToEdit = response[0];
-      modalRef.componentInstance.capabilities = this.capabilities;
-      modalRef.componentInstance.bands = this.bands;
-      modalRef.componentInstance.roleEdited.subscribe(data => {
-
-        this.showAlert = true;
-        if (data.data.hasOwnProperty('success')) {
-          this.alertMessage = data.data.success;
-          this.alertType = "success";
-        } else {
-          this.alertMessage = data.data.error;
-          this.alertType = "danger";
-        }
-        this.loadRoles();
-      });
-    });
+  displayAlert(data){
+    if(data.hasOwnProperty('success')){
+      this.showAlert = true;
```

Comment:
Call `this.showAlert = true;` once at the top of this function, as it's the same regardless if the `if` is true or false